### ISSUE TITLE
Implicit dependency on poison lib

### ISSUE THREAD

**Issue by pdgonzalez872** [kipcole9/money#53]
Hi!

I created a new mix project and was about to start playing around with this lib. However, the project failed to compile.

Here was the error:
```
==> ex_cldr
Compiling 2 files (.erl)
Compiling 31 files (.ex)

== Compilation error in file lib/cldr/config/config.ex ==
** (RuntimeError) The configured json library Poison does not define the function decode!/1
    lib/cldr/config/config.ex:119: (module)
    (stdlib) erl_eval.erl:670: :erl_eval.do_apply/6
could not compile dependency :ex_cldr, "mix compile" failed. You can recompile this dependency with "mix deps.compile ex_cldr", update it with "mix deps.update ex_cldr" or clean it with "mix deps.clean ex_cldr"
```

It looks like your lib has a dependency on `ex_cldr` which in turn has a dependency on https://github.com/devinus/poison.

Adding `poison` to the project resolved all issues. Maybe you'd like to make this dependency explicit somewhere, maybe here or maybe in `ex_cldr`. The latter is probably best. Thoughts?

**Comment by LostKobrakai** [comment]
The dependency is implicit (and documented) on both projects, because the json library is actually configurable by the enduser.

**Comment by pdgonzalez872** [comment]
Interesting! Thanks!

**Comment by kipcole9** [comment]
Thanks for the issue.  I've [pushed a commit to master](https://github.com/kipcole9/money/commit/a9ef234bdb24aae7494d8ea37c06f71ab89d83ae) that will print a more reasonable message if the configured job library (explicit or implicit) doesn't appear to be available.
```
$ iex -S mix
Erlang/OTP 20 [erts-9.2] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]

Compiling 5 files (.ex)

The json_library 'Jasonx' does not appear
to be available.  A json library is required
for Money to operate. Is in configured as a
dependency in mix.exs?

In config.exs your expicit or implicit configuration is:

  config ex_money
    json_library: Jasonx

In mix.exs you will need something like:

  def deps() do
    [
      ...
      {:jasonx, version_string}
    ]

== Compilation error in file lib/money.ex ==
** (ArgumentError) Json library jasonx does not appear to be a dependency
    lib/money.ex:71: (module)
    (stdlib) erl_eval.erl:670: :erl_eval.do_apply/6
```

**Comment by pdgonzalez872** [comment]
awesome!

**Comment by kipcole9** [comment]
Note that I have also [pushed a commit to Cldr master](https://github.com/kipcole9/cldr/commit/66cc8ace9cddcb31f030acb80dda78dba4f38d85) that will autodetect whether `Poison` or `Jason` are configured as dependencies and it will set the `:json_library accordingly.  `Jason` is preferred over `Poison` if it is configured.

Since `ex_money` uses `ex_cldr`'s default `:json_library` as its own default this will make overall configuration more straight forward and predictable.

A json library is still required to be configured as a dependency in `mix.exs`.  There's no real way around that requirement.  But having configured either `Poison` or `Jason` the rest should be automatic. 

This will be included in Cldr 1.4.1 which I expect to push to hex in the next 24 hours.

**Comment by kipcole9** [comment]
[Cldr version 1.4.1](https://hex.pm/packages/ex_cldr) has been pushed to hex, including the auto-detection of `Jason` and `Poison` json libraries.  That will also flow through to `ex_money`.  So if you have one of those libs configured in your `deps` then you'll be good to go.
